pkg/scalers/rabbitmq_scaler.go

```diff
@@ -130,8 +155,27 @@ func parseRabbitMQMetadata(resolvedEnv, metadata, authParams map[string]string)
 	return &meta, nil
 }
 
-func getConnectionAndChannel(host string) (*amqp.Connection, *amqp.Channel, error) {
-	conn, err := amqp.Dial(host)
+func getConnectionAndChannel(host string, caFile string, certFile string, keyFile string) (*amqp.Connection, *amqp.Channel, error) {
+	var conn *amqp.Connection
+	var err error
+
+	if strings.HasPrefix(host, amqps) {
+		cfg := new(tls.Config)
+
+		cfg.RootCAs = x509.NewCertPool()
+
+		if ca, err := ioutil.ReadFile(caFile); err == nil {
+			cfg.RootCAs.AppendCertsFromPEM(ca)
+		}
+
+		if cert, err := tls.LoadX509KeyPair(certFile, keyFile); err == nil {
```

Comment:
In my opinion, this should not be the required parameter. client cert and the key are only required why client authentication at the server is set to REQUIREANDVERIFY. In other cases, it could be empty. 

Until and unless I am missing something completely in regards to rabbitmq tls.